src/inspector_socket.cc

```diff
@@ -233,20 +232,25 @@ static ws_decode_result decode_frame_hybi17(const char* buffer_begin,
   }
   size_t payload_length = static_cast<size_t>(payload_length64);
 
-  if (data_length - kMaskingKeyWidthInBytes < payload_length)
+  if (buffer.size() - kMaskingKeyWidthInBytes < payload_length)
     return FRAME_INCOMPLETE;
 
-  const char* masking_key = p;
-  const char* payload = p + kMaskingKeyWidthInBytes;
+  std::vector<char>::const_iterator masking_key = it;
+  std::vector<char>::const_iterator payload = it + kMaskingKeyWidthInBytes;
   for (size_t i = 0; i < payload_length; ++i)  // Unmask the payload.
     output->insert(output->end(),
                    payload[i] ^ masking_key[i % kMaskingKeyWidthInBytes]);
 
-  size_t pos = p + kMaskingKeyWidthInBytes + payload_length - buffer_begin;
+  size_t pos = it + kMaskingKeyWidthInBytes + payload_length - buffer.begin();
   *bytes_consumed = pos;
   return closed ? FRAME_CLOSE : FRAME_OK;
 }
 
+inspector_socket_t* inspector_from_stream(void* client) {
+  return node::ContainerOf(&inspector_socket_t::client,
+                           reinterpret_cast<uv_tcp_t*>(client));
+}
```

Comment:
This could be made a little more type-safe by having overloads for uv_handle_t and uv_stream_t:
```c++
inspector_socket_t* inspector_from_tcp(uv_tcp_t* client) {
  return node::ContainerOf(&inspector_socket_t::client, client);
}
inspector_socket_t* inspector_from_stream(uv_stream_t* client) {
  return inspector_from_tcp(reinterpret_cast<uv_tcp_t*>(client));
}
inspector_socket_t* inspector_from_handle(uv_handle_t* client) {
  return inspector_from_tcp(reinterpret_cast<uv_tcp_t*>(client));
}
```